helix-term/src/commands/typed.rs

```diff
@@ -108,30 +114,115 @@ fn open(cx: &mut compositor::Context, args: &[Cow<str>], event: PromptEvent) ->
     }
 
     ensure!(!args.is_empty(), "wrong argument count");
+
+    fn open_file(cx: &mut compositor::Context, path: &Path, pos: Position) -> anyhow::Result<()> {
```

Comment:
Please don't make these inner functions. I find those quite hard to read with all the nesting